include/nuttx/spinlock.h

```diff
@@ -680,6 +703,8 @@ void spin_unlock_irqrestore(FAR volatile spinlock_t *lock,
 
 static inline_function void read_lock(FAR volatile rwlock_t *lock)
 {
+  sched_lock_wo_note();
```

Comment:
ALL need use the normal version:
```suggestion
  sched_lock();
```